src/server/src/main/java/io/cassandrareaper/ReaperApplication.java

```diff
@@ -332,13 +335,22 @@ private void maybeInitializeSidecarMode() throws ReaperException {
                 .getEnforcedLocalNode()
                 .orElse(clusterFacade.getLocalEndpoint(host));
         LOG.info("Sidecar mode. Local node is : {}", context.localNodeAddress);
+        selfRegisterClusterForSidecar(addClusterResource);
       } catch (RuntimeException | InterruptedException | ReaperException e) {
         LOG.error("Failed connecting to the local node in sidecar mode {}", host, e);
         throw new ReaperException(e);
       }
     }
   }
 
+  private void selfRegisterClusterForSidecar(ClusterResource addClusterResource) {
+    addClusterResource.addOrUpdateCluster(
+        Optional.empty(),
+        Optional.empty(),
+        Optional.of("127.0.0.1"),
+        Optional.of(Cluster.DEFAULT_JMX_PORT));
```

Comment:
```
  public boolean selfRegister(String seedHost, int jmxPort) {

    final Cluster cluster = findClusterWithSeedHost(seedHost, jmxPort);
    if (null == cluster) {
      return false; // shutdown reaper? no point in starting side-car when C* is not there and running first…
    }

    if (context.storage.getClusters().stream().noneMatch(c -> c.getName().equals(cluster.getName()))) {
      LOG.info("registering new cluster : {}", cluster.getName());

      // it is ok for this be called in parallel by different sidecars on the same cluster, so long we're not split-brain
      context.storage.addCluster(cluster);

      if (context.config.hasAutoSchedulingEnabled()) {
        try {
          clusterRepairScheduler.scheduleRepairs(cluster);
        } catch (ReaperException e) {
          LOG.error("failed to automatically schedule repairs for sidecar cluster {}", cluster.getName(), e);
          return false; // what to do in this situation for sidecar??
        }
      }
    }
    return true;
  }
```

Reply:
I don't see a benefit in duplicating some of the logic from `addOrUpdateCluster()` here.
Also, Cassandra could be down on the local node when Reaper starts, so it's fine to continue startup even if adding the cluster fails.
Auto scheduling will kick in eventually as well.

Reply:
> I don't see a benefit in duplicating some of the logic from addOrUpdateCluster() here.

`addOrUpdateCluster()` in its current form returns a Response and is intended for rest clients.
while this code is definitely not a rest client. there's value in having a more appropriate api signature. (dry can do as much damage as good sometimes.)

the original method also has additional and different logic to what is needed here: updating existing clusters is not required, and as you point out how to handle a down C* node, and/or auto-scheduling failing, is to be dealt with differently here.

Reply:
i'm easy which way you go, just pointing out the mismatches… 

i do think we need to be clear and decisive about what functionality we are re-using here, particularly that around the "updating an existing cluster with a new node list"